utils/load_path.ml

```diff
@@ -23,23 +23,26 @@ let files_uncap : registry = ref SMap.empty
 module Dir = struct
   type t = {
     path : string;
+    exists : bool;
     files : string list;
   }
 
   let path t = t.path
+  let exists t = t.exists
   let files t = t.files
 
   (* For backward compatibility reason, simulate the behavior of
      [Misc.find_in_path]: silently ignore directories that don't exist
      + treat [""] as the current directory. *)
   let readdir_compat dir =
     try
-      Sys.readdir (if dir = "" then Filename.current_dir_name else dir)
+      true, Sys.readdir (if dir = "" then Filename.current_dir_name else dir)
     with Sys_error _ ->
```

Comment:
Note that the original code hides permission errors. This is likely a bad idea but I didn't try to fix this in this PR (The code paths for `ocamldep` do not do this).